psi4/src/psi4/lib3index/dfhelper.cc

```diff
@@ -278,35 +278,34 @@ void DFHelper::prepare_sparsity() {
     std::vector<std::shared_ptr<TwoBodyAOInt>> eri(nthreads);
     std::vector<const double*> buffer(nthreads);
 
-    int rank = 0;
-#pragma omp parallel for private(rank) num_threads(nthreads) if (nao_ > 1000)
-    for (size_t i = 0; i < nthreads; i++) {
+#pragma omp parallel num_threads(nthreads) if (nao_ > 1000)
+    {
+        int rank = 0;
 #ifdef _OPENMP
         rank = omp_get_thread_num();
 #endif
         eri[rank] = std::shared_ptr<TwoBodyAOInt>(rifactory->eri());
         buffer[rank] = eri[rank]->buffer();
     }
 
-    double val, max_val = 0.0;
-    size_t MU, NU, mu, nu, omu, onu, nummu, numnu, index;
-#pragma omp parallel for private(MU, NU, mu, nu, omu, onu, nummu, numnu, index, val, \
-        rank) num_threads(nthreads) if (nao_ > 1000) schedule(guided) reduction(max:max_val)
-    for (MU = 0; MU < pshells_; ++MU) {
+    double max_val = 0.0;
+#pragma omp parallel for num_threads(nthreads) if (nao_ > 1000) schedule(guided) reduction(max:max_val)
+    for (int MU = 0; MU < pshells_; ++MU) {
+        int rank = 0;
 #ifdef _OPENMP
         rank = omp_get_thread_num();
 #endif
-        nummu = primary_->shell(MU).nfunction();
-        for (NU = 0; NU <= MU; ++NU) {
-            numnu = primary_->shell(NU).nfunction();
+        int nummu = primary_->shell(MU).nfunction();
+        for (int NU = 0; NU <= MU; ++NU) {
+            int numnu = primary_->shell(NU).nfunction();
             eri[rank]->compute_shell(MU, NU, MU, NU);
-            for (mu = 0; mu < nummu; ++mu) {
-                omu = primary_->shell(MU).function_index() + mu;
-                for (nu = 0; nu < numnu; ++nu) {
-                    onu = primary_->shell(NU).function_index() + nu;
+            for (int mu = 0; mu < nummu; ++mu) {
+                int omu = primary_->shell(MU).function_index() + mu;
+                for (int nu = 0; nu < numnu; ++nu) {
+                    int onu = primary_->shell(NU).function_index() + nu;
                     if (omu >= onu) {
-                        index = mu * (numnu * nummu * numnu + numnu) + nu * (nummu * numnu + 1);
-                        val = fabs(buffer[rank][index]);
+                        int index = mu * (numnu * nummu * numnu + numnu) + nu * (nummu * numnu + 1);
```

Comment:
@jgonthier makes a great point, this line will begin to fail for medium to large systems. Keeping this as `size_t` anywhere that does not interact with OMP would be preferable. Cannot believe I missed this.